docs/en/02_Developer_Guides/03_Forms/Field_types/05_UploadField.md

```diff
@@ -213,142 +272,152 @@ First create a [DataExtension](/developer_guides/extending/extensions) like this
-
- - `setAllowedFileExtensions`: (array) List of file extensions allowed.
-
- - `setAllowedFileCategories`: (array|string) List of types of files allowed. May be any of 'image', 'audio', 'mov', 'zip', 'flash', or 'doc'.
-
- - `setAutoUpload`: (boolean) Should the field automatically trigger an upload once a file is selected?
-
- - `setCanAttachExisting`: (boolean|string) Can the user attach existing files from the library. String values are interpreted as permission codes.
-
- - `setCanPreviewFolder`: (boolean|string) Can the user preview the folder files will be saved into? String values are interpreted as permission codes.
-
- - `setCanUpload`: (boolean|string) Can the user upload new files, or just select from existing files. String values are interpreted as permission codes.
-
- - `setDownloadTemplateName`: (string) javascript template used to display already uploaded files, see javascript/UploadField_downloadtemplate.js.
 
- - `setFileEditFields`: (FieldList|string) FieldList $fields or string $name (of a method on File to provide a fields) for the EditForm (Example: 'getCMSFields').
 
- - `setFileEditActions`: (FieldList|string) FieldList $actions or string $name (of a method on File to provide a actions) for the EditForm (Example: 'getCMSActions').
-
- - `setFileEditValidator`: (string) Validator (eg RequiredFields) or string $name (of a method on File to provide a Validator) for the EditForm (Example: 'getCMSValidator').
-
- - `setOverwriteWarning`: (boolean) Show a warning when overwriting a file.
+Then, in your GalleryPage, tell the UploadField to use this function:
 
- - `setPreviewMaxWidth`: (int).
 
- - `setPreviewMaxHeight`: (int).
+	:::php
+	$uploadField->setFileEditFields('getCustomFields');
 
- - `setTemplateFileButtons`: (string) Template name to use for the file buttons.
 
- - `setTemplateFileEdit`: (string) Template name to use for the file edit form.
+In a similar fashion you can use 'setFileEditActions' to set the actions for the editform, or
+'fileEditValidator' to determine the validator (e.g. RequiredFields). 
 
- - `setUploadTemplateName`: (string) javascript template used to display uploading files, see javascript/UploadField_uploadtemplate.js.
+### Configuration Reference
 
- - `setCanPreviewFolder`: (boolean|string) Is the upload folder visible to uploading users? String values are interpreted as permission codes.
+ * `setAllowedMaxFileNumber`: (int) php validation of allowedMaxFileNumber only works when a db 
+   relation is available, set to null to allow unlimited if record has a has_one and
+   allowedMaxFileNumber is null, it will be set to 1.
+ * `setAllowedFileExtensions`: (array) List of file extensions allowed.
+ * `setAllowedFileCategories`: (array|string) List of types of files allowed. May be any of
```

Comment:
`setAllowedFileCategories` internally maps the given categories to extensions, then invokes `setAllowedFileExtensions`